### ISSUE TITLE
Run CI on Perlmutter

### ISSUE THREAD

**Issue by shahzebsiddiqui** [buildtesters/buildtest#1301]
Action Items
- [x] Rename https://github.com/buildtesters/buildtest/blob/devel/.gitlab/cori.yml file to something like `nersc.yml` 
- [x] Rename https://github.com/buildtesters/buildtest/blob/devel/tests/test_cori.py to `test_nersc.py` and add method to run on Perlmutter
- [x] Make use of new gitlab runner to run jobs via user [bdtest](https://iris.nersc.gov/user/99914/info)

**Comment by jscook2345** [comment]
This looks like a duplicate of #1302, could these issues be combined if you're just going to make it a big checklist?

**Comment by shahzebsiddiqui** [comment]
So this task depends on #1302 and will require a fair bit of development, code-review whereas #1301 is a special type of issue that doesn't require any contribution to buildtest. It just requires setting up the gitlab runner/jacamar so that we can perform this task. Typically all issues have contribution to project that can be linked with MR, but in this case we just need to setup runner.

I'd prefer these tasks be separate to make it easier for @prathmesh4321 to work on smaller set of changes.
